src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,557 @@
+    }
+
+    private MessageFormat generateMessageFormat(Object[] input) {
+        var len = input.length;
+        return switch (len) {
+            case 0 -> throw new IllegalArgumentException("There should at least be one input string");
+            case 1 -> new MessageFormat("{0}", locale);
+            case 2, 3 -> {
+                var pattern = patterns[len + 1];
+                if (pattern != null && !pattern.isEmpty()) {
+                    yield new MessageFormat(pattern, locale);
+                } else {
+                    yield new MessageFormat(createMessageFormatString(len), locale);
+                }
+            }
+            default -> new MessageFormat(createMessageFormatString(len), locale);
+        };
+    }
+
+    private String createMessageFormatString(int count) {
+        var sb = new StringBuilder(patterns[START]);
+        IntStream.range(2, count - 1).forEach(i -> sb.append(middleBetween).append("{").append(i).append("}"));
+        sb.append(patterns[END].replaceFirst("\\{0}", "").replaceFirst("\\{1}", "\\{" + (count - 1) + "\\}"));
+        return sb.toString();
+    }
+
+    @java.io.Serial
+    private void readObject(ObjectInputStream stream)
+            throws IOException, ClassNotFoundException {
+        stream.defaultReadObject();
+        init();
+    }
+
+    /**
+     * A list format type - {@code standard}, {@code or}, and {@code unit}.
+     * <p>
+     * {@code Type} is an enum which represents the type for formatting
+     * a list within a given {@code ListFormat} instance.
+     */
+    public enum Type {
+
+        /**
+         * The {@code STANDARD} list format style. This is the default
+         * type, which concatenates elements in "and" enumeration.
+         */
+        STANDARD,
+
+        /**
+         * The {@code OR} list format style. This style concatenates
+         * elements in "or" enumeration.
+         */
+        OR,
+
+        /**
+         * The {@code UNIT} list format style. This style concatenates
+         * elements, useful for enumerating units.
+         */
+        UNIT;
+
+        @Override
+        public String toString() {
+            return name().toLowerCase(Locale.ROOT);
+        }
+    }
+
+    /**
+     * A list format style - {@code full}, {@code short}, and {@code narrow}.
+     * <p>
+     * {@code Style} is an enum which represents the style for formatting
+     * a list within a given {@code ListFormat} instance.
+     */
+    public enum Style {
+
+        /**
+         * The {@code FULL} list format style.
+         */
+        FULL,
+
+        /**
+         * The {@code SHORT} list format style.
+         */
+        SHORT,
+
+        /**
+         * The {@code NARROW} list format style.
+         */
+        NARROW;
+
+        @Override
+        public String toString() {
+            return name().toLowerCase(Locale.ROOT);
```

Comment:
Don't override to lower case, leave the java enum alone.  It will need to be UPPERCASE in the source code.